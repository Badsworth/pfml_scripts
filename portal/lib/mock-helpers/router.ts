/**
 * @file Common next/router mocks for use in Storybook and Tests
 */
import Router from "next/router";
import createMockFn from "./createMockFn";

export interface MockRouterEvent {
  name: string;
  callback: RouterEventCallback;
}

export const mockRouterEvents: MockRouterEvent[] = [];

type RouterEventCallback = (...args: unknown[]) => unknown;

// @ts-expect-error Router mock is missing properties we don't currently reference
export const mockRouter = {
  locale: "en-US",
  route: "/",
  pathname: "",
  query: {},
  asPath: "",
  back: createMockFn(),
  beforePopState: createMockFn(),
  prefetch: createMockFn(() => Promise.resolve()),
  push: createMockFn(),
  replace: createMockFn(),
  reload: createMockFn(),
  events: {
    on: (name: string, callback: RouterEventCallback) => {
      mockRouterEvents.push({ name, callback });
    },
    off: createMockFn(),
  },
  isFallback: false,
} as typeof Router.router;

export function useRouter() {
  return mockRouter;
}
