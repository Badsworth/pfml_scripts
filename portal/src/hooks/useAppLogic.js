/* eslint sort-keys: ["error", "asc"] */
import useAppErrorsLogic from "./useAppErrorsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useFlagsLogic from "./useFeatureFlagsLogic";
import useOtherLeavesLogic from "./useOtherLeavesLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const { appErrors, ...appErrorsLogic } = useAppErrorsLogic({ portalFlow });
  const auth = useAuthLogic({ appErrorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    appErrorsLogic,
    isLoggedIn: auth.isLoggedIn,
    portalFlow,
  });

  // user will be eventually set through a `login` method
  // const [user, setUser] = useState();
  // TODO (CP-885): remove user from configs and write login method
  const benefitsApplications = useBenefitsApplicationsLogic({
    appErrorsLogic,
    portalFlow,
    user: users.user,
  });

  const claims = useClaimsLogic({
    appErrorsLogic,
  });

  const documents = useDocumentsLogic({
    appErrorsLogic,
    portalFlow,
  });

  const employers = useEmployersLogic({
    appErrorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const otherLeaves = useOtherLeavesLogic({ appErrorsLogic });
  const featureFlags = useFlagsLogic({ appErrorsLogic });

  return {
    appErrors,
    auth,
    benefitsApplications,
    catchError: appErrorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: appErrorsLogic.clearErrors,
    documents,
    employers,
    otherLeaves,
    portalFlow,
    setAppErrors: appErrorsLogic.setAppErrors,
    users,
    featureFlags,
  };
};

export default useAppLogic;
