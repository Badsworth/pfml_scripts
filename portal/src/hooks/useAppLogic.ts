/* eslint sort-keys: ["error", "asc"] */
import useApplicationImportsLogic from "./useApplicationImportsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitYearsLogic from "./useBenefitYearsLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useErrorsLogic from "./useErrorsLogic";
import useFeatureFlagsLogic from "./useFeatureFlagsLogic";
import useHolidaysLogic from "./useHolidaysLogic";
import usePaymentsLogic from "./usePaymentsLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const errorsLogic = useErrorsLogic({ portalFlow });

  const applicationImports = useApplicationImportsLogic({
    errorsLogic,
    portalFlow,
  });
  const auth = useAuthLogic({ errorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    errorsLogic,
    isLoggedIn: !!auth.isLoggedIn,
    portalFlow,
  });

  const benefitYears = useBenefitYearsLogic({ errorsLogic });

  const benefitsApplications = useBenefitsApplicationsLogic({
    errorsLogic,
    portalFlow,
  });

  const claims = useClaimsLogic({
    errorsLogic,
    portalFlow,
  });

  const documents = useDocumentsLogic({
    errorsLogic,
  });

  const employers = useEmployersLogic({
    clearClaims: claims.clearClaims,
    errorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const featureFlags = useFeatureFlagsLogic();

  const holidays = useHolidaysLogic({
    errorsLogic,
  });

  const payments = usePaymentsLogic({
    errorsLogic,
  });

  return {
    // `_errorsLogic` should not be used except for testing
    _errorsLogic: errorsLogic,
    applicationImports,
    auth,
    benefitYears,
    benefitsApplications,
    catchError: errorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: errorsLogic.clearErrors,
    clearRequiredFieldErrors: errorsLogic.clearRequiredFieldErrors,
    documents,
    employers,
    errors: errorsLogic.errors,
    featureFlags,
    holidays,
    payments,
    portalFlow,
    setErrors: errorsLogic.setErrors,
    users,
  };
};

export default useAppLogic;
export type AppLogic = ReturnType<typeof useAppLogic>;
