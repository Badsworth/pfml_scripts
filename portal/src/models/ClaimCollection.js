import BaseCollection from "./BaseCollection";
import { ClaimStatus } from "./Claim";

/** @typedef {import('./Claim').default} Claim */

export default class ClaimCollection extends BaseCollection {
  get idProperty() {
    return "application_id";
  }

  /**
   * @returns {Claim[]} Returns all claims with an "In progress" status
   */
  get inProgress() {
    return this.items.filter((item) => item.status === ClaimStatus.started);
  }

  /**
   * @returns {Claim[]} Returns all claims submitted to the API, including
   * those that made it to the Claims Processing System
   */
  get submitted() {
    return this.items.filter((item) => item.isCompleted || item.isSubmitted);
  }
}
