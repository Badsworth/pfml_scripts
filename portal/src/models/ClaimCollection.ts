import BaseCollection from "./BaseCollection";
import Claim from "./Claim";

export default class ClaimCollection extends BaseCollection<Claim> {
  get idProperty() {
    return "fineos_absence_id";
  }
}
