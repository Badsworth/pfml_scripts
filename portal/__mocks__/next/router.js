import { mockRouter, mockRouterEvents } from "../../lib/mock-helpers/router";
export * from "../../lib/mock-helpers/router";

afterEach(() => {
  mockRouter.pathname = "";

  // Clear mock router events
  mockRouterEvents.length = 0;
});
