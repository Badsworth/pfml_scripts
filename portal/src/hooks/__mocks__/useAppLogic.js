import Claim from "../../models/Claim";
import ClaimCollection from "../../models/ClaimCollection";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  auth: {
    createAccount: jest.fn(),
    forgotPassword: jest.fn(),
    login: jest.fn(),
  },
  claims: new ClaimCollection(),
  createClaim: jest.fn(() => new Claim({ application_id: uniqueId() })),
  loadClaims: jest.fn(),
  submitClaim: jest.fn(),
  updateClaim: jest.fn(),
}));
