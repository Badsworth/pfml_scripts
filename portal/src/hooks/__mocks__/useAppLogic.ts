import {
  BenefitsApplicationDocument,
  ClaimDocument,
} from "../../models/Document";
import User, { UserLeaveAdministrator } from "../../models/User";
import ApiResourceCollection from "../../models/ApiResourceCollection";
import BenefitsApplication from "../../models/BenefitsApplication";
import Claim from "../../models/Claim";
import ClaimDetail from "../../models/ClaimDetail";
import EmployerClaim from "../../models/EmployerClaim";
import Flag from "../../models/Flag";
import { Payment } from "src/models/Payment";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  errors: [],
  auth: {
    createAccount: jest.fn(),
    createEmployerAccount: jest.fn(),
    forgotPassword: jest.fn(),
    isLoggedIn: true,
    login: jest.fn(),
    logout: jest.fn(),
    requireLogin: jest.fn(),
    resendForgotPasswordCode: jest.fn(),
    resendVerifyAccountCode: jest.fn(),
    resetPassword: jest.fn(),
    verifyAccount: jest.fn(),
  },
  catchError: jest.fn(),
  benefitsApplications: {
    benefitsApplications: new ApiResourceCollection<BenefitsApplication>(
      "application_id"
    ),
    complete: jest.fn(),
    create: jest.fn(
      () => new BenefitsApplication({ application_id: uniqueId() })
    ),
    get: jest.fn(),
    hasLoadedBenefitsApplicationAndWarnings: jest.fn().mockReturnValue(true),
    hasLoadedAll: false,
    load: jest.fn(),
    loadAll: jest.fn(),
    submit: jest.fn(),
    submitPaymentPreference: jest.fn(),
    update: jest.fn(),
    warningsLists: {},
  },
  claims: {
    claimDetail: new ClaimDetail(),
    claims: new ApiResourceCollection<Claim>("fineos_absence_id"),
    clearClaims: jest.fn(),
    isLoadingClaims: null,
    isLoadingClaimDetail: null,
    loadClaimDetail: jest.fn(),
    paginationMeta: {},
    loadPage: jest.fn(),
  },
  clearErrors: jest.fn(),
  clearRequiredFieldErrors: jest.fn(),
  documents: {
    attach: jest.fn((application_id, files) => {
      const uploadPromises: Array<Promise<{ success: boolean }>> = [];
      for (let i = 0; i < files.length; i++) {
        uploadPromises.push(Promise.resolve({ success: true }));
      }
      return uploadPromises;
    }),
    hasLoadedClaimDocuments: jest.fn(),
    documents: new ApiResourceCollection<BenefitsApplicationDocument>(
      "fineos_document_id"
    ),
    download: jest.fn(),
    loadAll: jest.fn(),
  },
  employers: {
    addEmployer: jest.fn(),
    downloadDocument: jest.fn(() => new Blob()),
    loadClaim: jest.fn(
      () =>
        new EmployerClaim({
          absence_periods: [],
          fineos_absence_id: "NTN-111-ABS-01",
        })
    ),
    loadDocuments: jest.fn(
      () => new ApiResourceCollection<ClaimDocument>("fineos_document_id")
    ),
    loadWithholding: jest.fn(() => ({ filing_period: "2011-11-20" })),
    submitClaimReview: jest.fn(),
    submitWithholding: jest.fn(),
  },
  payments: {
    loadPayments: jest.fn(),
    loadedPaymentsData: new Payment(),
    hasLoadedPayments: true,
    isLoadingPayments: false,
  },
  portalFlow: {
    getNextPageRoute: jest.fn(),
    goTo: jest.fn(),
    goToNextPage: jest.fn(),
    goToPageFor: jest.fn(),
    pathname: "/mock-portalFlow-pathname",
  },
  otherLeaves: {
    removeEmployerBenefit: jest.fn(() => true),
    removeOtherIncome: jest.fn(() => true),
    removePreviousLeave: jest.fn(() => true),
  },
  setErrors: jest.fn(),
  updateUser: jest.fn(),
  user: new User({
    user_id: "mock_user_id",
    consented_to_data_sharing: true,
    email_address: "ali@miau.com",
    auth_id: "cognito_123",
    roles: [],
    user_leave_administrators: [],
  }),
  users: {
    loadUser: jest.fn(),
    requireUserConsentToDataAgreement: jest.fn(),
    requireUserRole: jest.fn(),
    updateUser: jest.fn(),
    user: new User({
      auth_id: "mock_auth_id",
      user_id: "mock_user_id",
      consented_to_data_sharing: true,
      user_leave_administrators: [
        // not verified, but can be
        new UserLeaveAdministrator({
          employer_dba: "Book Bindings 'R Us",
          employer_fein: "**-***1823",
          employer_id: "dda903f-f093f-ff900",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: false,
        }),
        // already verified
        new UserLeaveAdministrator({
          employer_dba: "Knitting Castle",
          employer_fein: "**-***3443",
          employer_id: "dda930f-93jfk-iej08",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: true,
        }),
        // not verified and cannot be verified
        new UserLeaveAdministrator({
          employer_dba: "Tomato Touchdown",
          employer_fein: "**-***7192",
          employer_id: "io19fj9-00jjf-uiw3r",
          has_fineos_registration: true,
          has_verification_data: false,
          verified: false,
        }),
      ],
    }),
  },
  featureFlags: {
    flags: [
      new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      }),
    ],
    getFlag: jest.fn(() => {
      return new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      });
    }),
    loadFlags: jest.fn(),
  },
}));
