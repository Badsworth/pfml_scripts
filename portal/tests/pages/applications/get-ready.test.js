import {
  MockClaimBuilder,
  renderWithAppLogic,
  testHook,
} from "../../test-utils";
import ClaimCollection from "../../../src/models/ClaimCollection";
import GetReady from "../../../src/pages/applications/get-ready";
import User from "../../../src/models/User";
import { mockRouter } from "next/router";
import routes from "../../../src/routes";
import useAppLogic from "../../../src/hooks/useAppLogic";

function render(customProps = {}, options = { claims: [] }) {
  let appLogic;
  mockRouter.pathname = routes.applications.getReady;

  testHook(() => {
    appLogic = useAppLogic();
    appLogic.users.user = new User({ consented_to_data_sharing: true });
    appLogic.claims.claims = new ClaimCollection(options.claims);
    appLogic.claims.hasLoadedAll = true;
  });

  const { wrapper } = renderWithAppLogic(GetReady, {
    props: { appLogic, ...customProps },
  });

  return { appLogic, wrapper };
}

describe("GetReady", () => {
  it("renders get ready content", () => {
    const { wrapper } = render();

    expect(wrapper).toMatchSnapshot();

    wrapper.find("Trans").forEach((trans) => {
      expect(trans.dive()).toMatchSnapshot();
    });
  });

  it("doesn't show link to applications when claims do not exist", () => {
    const { wrapper } = render();

    expect(wrapper.find("Link").exists()).toBe(false);
  });

  it("shows link to applications when claims exist", () => {
    const claims = [new MockClaimBuilder().create()];
    const { wrapper } = render({}, { claims });

    expect(wrapper.find("Link")).toMatchInlineSnapshot(`
      <Link
        href="/applications"
      >
        <a
          className="display-inline-block margin-bottom-5"
        >
          View all applications
        </a>
      </Link>
    `);
  });
});
