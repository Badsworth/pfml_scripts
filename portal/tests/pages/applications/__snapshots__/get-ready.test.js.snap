// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetReady renders get ready content 1`] = `
<Fragment>
  <Title>
    Get ready to apply
  </Title>
  <Alert
    className="margin-bottom-3"
    heading="Only some people can apply online for now"
    neutral={true}
    state="info"
  >
    <Heading
      className="margin-top-3"
      level="3"
    >
      <SvgIconLaptop
        className="margin-right-1 text-secondary text-middle margin-top-neg-05"
        fill="currentColor"
        height={20}
        width={20}
      />
      Apply online
    </Heading>
    <Trans
      components={
        Object {
          "li": <li />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.getReady.alertOnline"
      tOptions={
        Object {
          "context": undefined,
        }
      }
    />
    <Heading
      level="3"
    >
      <SvgIconPhone
        className="margin-right-1 text-secondary text-middle margin-top-neg-05"
        fill="currentColor"
        height={20}
        width={20}
      />
      Apply by phone
    </Heading>
    <Trans
      components={
        Object {
          "benefits-timeline-link": <a
            href="https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-timeline#december-2,-2020-"
          />,
          "contact-center-phone-link": <a
            href="tel:(833) 344-7365"
          />,
          "li": <li />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.getReady.alertPhone"
    />
  </Alert>
  <div
    className="measure-6"
  >
    <Heading
      level="2"
    >
      <Icon
        className="margin-right-1 text-secondary text-middle margin-top-neg-05"
        name="comment"
      />
      1. Tell your employer that you need to take paid leave
    </Heading>
    <Trans
      i18nKey="pages.getReady.stepOne"
    />
    <Heading
      level="2"
    >
      <Icon
        className="margin-right-1 text-secondary text-middle margin-top-neg-05"
        name="upload_file"
      />
      2. Get documentation that supports your leave request
    </Heading>
    <Trans
      components={
        Object {
          "which-paid-leave-link": <a
            href="https://www.mass.gov/which-paid-leave"
            rel="noopener"
            target="_blank"
          />,
        }
      }
      i18nKey="pages.getReady.stepTwoWhichPaidLeaveBody"
    />
    <Heading
      level="3"
    >
      Medical leave for your own serious health condition
    </Heading>
    <Trans
      components={
        Object {
          "healthcare-provider-form-link": <a
            href="https://www.mass.gov/medical-leave-form"
          />,
        }
      }
      i18nKey="pages.getReady.stepTwoMedicalLeaveBody"
    />
    <Heading
      level="3"
    >
      Family leave
    </Heading>
    <Heading
      level="4"
    >
      To bond with a child
    </Heading>
    <Trans
      i18nKey="pages.getReady.stepTwoBondingLeaveBody"
    />
    <Heading
      level="2"
    >
      <Icon
        className="margin-right-1 text-secondary text-middle margin-top-neg-05"
        name="edit"
      />
      3. Apply
    </Heading>
    <Trans
      components={
        Object {
          "contact-center-phone-link": <a
            href="tel:(833) 344-7365"
          />,
          "li": <li />,
          "medical-bonding-link": <a
            href="https://www.mass.gov/pfml-medical-bonding"
          />,
          "tax-liability-link": <a
            href="https://www.mass.gov/pfml-tax-liability"
          />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.getReady.stepThree"
    />
    <ButtonLink
      className="margin-top-3 margin-bottom-8"
      href="/applications/start"
    >
      Create an application
    </ButtonLink>
  </div>
</Fragment>
`;

exports[`GetReady renders get ready content 2`] = `
Array [
  <p
    key="p-0"
  >
    If you are currently employed in Massachusetts but not self-employed, you can apply online or by phone for the following types of paid leave:
  </p>,
  <ul
    className="usa-list"
    key="1"
  >
    <li
      key="0"
    >
      Medical leave due to your own illness, injury, or pregnancy
    </li>
    <li
      key="1"
    >
      Family leave to bond with your child after birth, adoption, or foster placement — whether you are applying before or after the child arrives
    </li>
  </ul>,
]
`;

exports[`GetReady renders get ready content 3`] = `
Array [
  <p
    key="p-0"
  >
    Apply by calling the Department of Family and Medical Leave Contact Center at 
    <a
      href="tel:(833) 344-7365"
      key="1"
    >
      (833) 344‑7365
    </a>
     if 
    <strong
      key="strong-3"
    >
      any
    </strong>
     of the following are true:
  </p>,
  <ul
    className="usa-list"
    key="1"
  >
    <li
      key="0"
    >
      You are self-employed or unemployed and you are applying for paid leave
    </li>
    <li
      key="1"
    >
      You need paid family leave to care for a family member who serves in the armed forces
    </li>
    <li
      key="2"
    >
      You need paid family leave to manage family affairs when a family member is on active duty in the armed forces
    </li>
  </ul>,
]
`;

exports[`GetReady renders get ready content 4`] = `
Array [
  <p
    key="p-0"
  >
    If you can, tell your employer at least 30 days before your leave begins. If you need to take leave right away, tell your employer as soon as possible.
  </p>,
  <p
    key="p-1"
  >
    Once you tell your employer, you have the right to apply and your job is protected. Make a note of when you notified your employer. You will need to provide this date in your leave application.
  </p>,
]
`;

exports[`GetReady renders get ready content 5`] = `
<p
  key="p-0"
>
  If you’re not sure what kind of leave to apply for, 
  <a
    href="https://www.mass.gov/which-paid-leave"
    key="1"
    rel="noopener"
    target="_blank"
  >
    learn more about Paid Family and Medical Leave
  </a>
  .
</p>
`;

exports[`GetReady renders get ready content 6`] = `
<p
  key="p-0"
>
  Your health care provider must complete the 
  <a
    href="https://www.mass.gov/medical-leave-form"
    key="1"
  >
    Certification of Your Serious Health Condition
  </a>
  .
</p>
`;

exports[`GetReady renders get ready content 7`] = `
Array [
  <p
    key="p-0"
  >
    For family leave to bond with your child after birth, foster placement, or adoption you need a document that confirms your child’s date of birth or placement.
  </p>,
  <p
    key="p-1"
  >
    You can apply before your child is born or arrives in your home. You will need to provide proof of birth or placement for your application to be approved.
  </p>,
]
`;

exports[`GetReady renders get ready content 8`] = `
Array [
  <p
    key="p-0"
  >
    Applying takes around 15 minutes. Your information will save as you go, so you can finish your application later if you need to.
  </p>,
  <p
    key="p-1"
  >
    If you give birth and plan to take both pregnancy-related medical leave and family leave to bond with your newborn, you should apply for medical leave first. Family leave to bond with your child can be 
    <a
      href="https://www.mass.gov/pfml-medical-bonding"
      key="1"
    >
      easily added to your claim
    </a>
     by calling our Contact Center at 
    <a
      href="tel:(833) 344-7365"
      key="3"
    >
      (833) 344‑7365
    </a>
    .
  </p>,
  <p
    key="p-2"
  >
    You need to create multiple leave applications if you are:
  </p>,
  <ul
    className="usa-list"
    key="3"
  >
    <li
      key="0"
    >
      Taking leave from multiple employers.
    </li>
    <li
      key="1"
    >
      Taking time off in uneven blocks of time (intermittent leave), 
      <strong
        key="strong-1"
      >
        and
      </strong>
       taking time off completely or on a reduced schedule. You’ll need a separate application for the intermittent leave.
    </li>
  </ul>,
  <p
    key="p-4"
  >
    PFML benefits are subject to reporting for tax purposes and may be subject to taxation. Withholding is not currently supported through the PFML program. Learn more about the 
    <a
      href="https://www.mass.gov/pfml-tax-liability"
      key="1"
    >
      possible tax implications
    </a>
     of your paid leave benefits.
  </p>,
]
`;
